internal/campaigns/volume_workspace_test.go

```diff
@@ -39,103 +40,241 @@ func TestVolumeWorkspaceCreator(t *testing.T) {
-			),
-			expect.NewGlob(
-				expect.Behaviour{},
-				"docker", "run", "--rm", "--init", "--workdir", "/work",
-				"--mount", "type=bind,source=*,target=/tmp/zip,ro",
-				"--mount", "type=volume,source="+volumeID+",target=/work",
-				dockerVolumeWorkspaceImage,
-				"unzip", "/tmp/zip",
-			),
-			expect.NewGlob(
-				expect.Behaviour{},
-				"docker", "run", "--rm", "--init", "--workdir", "/work",
-				"--mount", "type=bind,source=*,target=/run.sh,ro",
-				"--mount", "type=volume,source="+volumeID+",target=/work",
-				dockerVolumeWorkspaceImage,
-				"sh", "/run.sh",
-			),
-		)
-
-		if w, err := wc.Create(ctx, repo, zip); err != nil {
-			t.Errorf("unexpected error: %v", err)
-		} else if have := w.(*dockerVolumeWorkspace).volume; have != volumeID {
-			t.Errorf("unexpected volume: have=%q want=%q", have, volumeID)
-		}
-	})
-
-	t.Run("docker volume create failure", func(t *testing.T) {
-		expect.Commands(
-			t,
-			expect.NewGlob(
-				expect.Behaviour{ExitCode: 1},
-				"docker", "volume", "create",
-			),
-		)
-
-		if _, err := wc.Create(ctx, repo, zip); err == nil {
-			t.Error("unexpected nil error")
-		}
-	})
-
-	t.Run("unzip failure", func(t *testing.T) {
-		expect.Commands(
-			t,
-			expect.NewGlob(
-				expect.Behaviour{Stdout: []byte(volumeID)},
-				"docker", "volume", "create",
-			),
-			expect.NewGlob(
-				expect.Behaviour{ExitCode: 1},
-				"docker", "run", "--rm", "--init", "--workdir", "/work",
-				"--mount", "type=bind,source=*,target=/tmp/zip,ro",
-				"--mount", "type=volume,source="+volumeID+",target=/work",
-				dockerVolumeWorkspaceImage,
-				"unzip", "/tmp/zip",
-			),
-		)
-
-		if _, err := wc.Create(ctx, repo, zip); err == nil {
-			t.Error("unexpected nil error")
-		}
-	})
-
-	t.Run("git init failure", func(t *testing.T) {
-		expect.Commands(
-			t,
-			expect.NewGlob(
-				expect.Behaviour{Stdout: []byte(volumeID)},
-				"docker", "volume", "create",
-			),
-			expect.NewGlob(
-				expect.Behaviour{},
-				"docker", "run", "--rm", "--init", "--workdir", "/work",
-				"--mount", "type=bind,source=*,target=/tmp/zip,ro",
-				"--mount", "type=volume,source="+volumeID+",target=/work",
-				dockerVolumeWorkspaceImage,
-				"unzip", "/tmp/zip",
-			),
-			expect.NewGlob(
-				expect.Behaviour{ExitCode: 1},
-				"docker", "run", "--rm", "--init", "--workdir", "/work",
-				"--mount", "type=bind,source=*,target=/run.sh,ro",
-				"--mount", "type=volume,source="+volumeID+",target=/work",
-				dockerVolumeWorkspaceImage,
-				"sh", "/run.sh",
-			),
-		)
-
-		if _, err := wc.Create(ctx, repo, zip); err == nil {
-			t.Error("unexpected nil error")
-		}
-	})
+	for name, tc := range map[string]struct {
+		expectations []*expect.Expectation
+		steps        []Step
+		wantErr      bool
+	}{
+		"no steps": {
+			expectations: []*expect.Expectation{
+				expect.NewGlob(
+					expect.Behaviour{Stdout: []byte(volumeID)},
+					"docker", "volume", "create",
+				),
+				expect.NewGlob(
+					expect.Behaviour{},
```

Comment:
I think a `var Noop = expect.Behaviour{}` in `expect` would be cool:

```suggestion
					expect.Noop,
```

Or `expect.NoOutput` or something.

Reply:
That's fair. `Noop` probably isn't quite right, but that probably reinforces why `Behaviour{}` is potentially misleading. I'll add a `Success` constant, I think.

Reply:
Ah, `Success` is good!